src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/criteria/Criteria.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.cosmosdb.core.criteria;
+
+import lombok.Getter;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+public class Criteria {
+
+    private String subject;
+    private List<Object> subValues;
+    private final CriteriaType type;
+    private final List<Criteria> subCriteria;
+
+    private Criteria(CriteriaType type) {
+        this.type = type;
+        this.subCriteria = new ArrayList<>();
+    }
+
+    public static boolean isBinaryOperation(CriteriaType type) {
+        switch (type) {
+            case AND:
+            case OR:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    public static boolean isUnaryOperation(CriteriaType type) {
```

Comment:
Does the `isBinaryOperation` and `isUnaryOperation` will be used outside this Criteria class?

Reply:
Yes, generator will use this when traversal Criteria tree.